src/server/snapshot.cc

```diff
@@ -54,9 +54,11 @@ void SliceSnapshot::Start(bool stream_journal) {
   sfile_.reset(new io::StringFile);
   rdb_serializer_.reset(new RdbSerializer(sfile_.get()));
 
-  snapshot_fb_ = fiber([this, stream_journal] {
-    SerializeEntriesFb();
-    if (!stream_journal) {
+  snapshot_fb_ = fiber([this, stream_journal, &cll] {
```

Comment:
i prefer that you pass the Cancellation object as a pointer (into the function). I find it confusing when constref is passed into an async lambda.

Reply:
Everywhere? So it'll be `const Cancellation* ccl`?

I made it a reference because the styleguide says so. Feels more natural, not sure what is confusing about the lambda? Accidental copy?

Reply:
Its hard to use this type incorrectly because atomic is neither movable neither copyable

Reply:
In that sense its true. Fixed it